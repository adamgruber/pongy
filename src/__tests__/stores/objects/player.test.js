import Player from '../../../stores/objects/player';

describe('Player', () => {
  let subject;

  beforeEach(() => {
    subject = new Player('1234', 'PlayerOne');
  });

  describe('constructor', () => {
    it('should create a Player instance with initial properties', () => {
      expect(subject).toBeInstanceOf(Player);
      expect(subject).toMatchSnapshot();
    });
  });

  describe('computed properties', () => {
  });

  describe('actions', () => {
    describe('incrementScore', () => {
      beforeEach(() => {
        subject.incrementScore();
      });

      it('should increment score by 1', () => {
        expect(subject.score).toBe(1);
      });
    });

    describe('decrementScore', () => {
      it('should NOT decrement score if it is 0', () => {
        subject.decrementScore();
        expect(subject.score).toBe(0);
      });

      it('should decrement score by 1', () => {
        subject.score = 3;
        subject.decrementScore();
        expect(subject.score).toBe(2);
      });
    });
  });
});
