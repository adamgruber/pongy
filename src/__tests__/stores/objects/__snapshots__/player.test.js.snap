// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player constructor should create a Player instance with initial properties 1`] = `
Player {
  "decrementScore": [Function],
  "id": "1234",
  "incrementScore": [Function],
  "score": 0,
  "title": "PlayerOne",
}
`;
